### DISCUSSION TITLE
v7.4.0 Long Pauses

### DISCUSSION THREAD

**Discussion by Tribo-2dot0** [arkypita/LaserGRBL#2268] [General]
Testei por duas vezes e em ambas o programa faz uma longa pausa no momento da transição do arquivo de gravação para o arquivo de corte, nesse momento o buffer fica vermelho; o aviso do Telegram soa avisando que o programa parou de responder... esperando um tempo (1 minuto, talvez um pouco menos) ele continua normalmente, mas ainda faz algumas pausas mais longas em determinados momentos (dessa vez sem aviso do Telegram; Me parece ser sempre antes de iniciar um caminho ou sequencia de caminhos retos), mas acaba dando continuidade e termina reportando 0 erros.

Aqui o sessionlog:
08/06/2024 13:57:11.067	Program     	------- LaserGRBL v7.4.0 [64bit] START -------
08/06/2024 13:57:11.600	OpenGL      	FBO OpenGL 4.6.0 NVIDIA 555.99, NVIDIA Corporation, NVIDIA GeForce RTX 3060/PCIe/SSE2
08/06/2024 13:57:24.002	OpenFile    	Open C:\Users\ferna\OneDrive\Desktop\Tambor de lingua\Tambor de Lingua_Tampo_Grav.png
08/06/2024 13:57:49.145	OpenFile    	Open C:\Users\ferna\OneDrive\Desktop\Tambor de lingua\Tambor de Lingua_Tampo_Cut.svg
08/06/2024 13:57:53.518	SetStatus   	Machine status [Connecting]
08/06/2024 13:57:53.534	SendImmediate	Send Immediate Command [0x18]
08/06/2024 13:57:53.592	VersionInfo 	Detected Grbl v1.1#
08/06/2024 13:57:53.596	SetStatus   	Machine status [Idle]
08/06/2024 13:57:54.100	Refresh Config	Refreshing grbl configuration [OnConnect]
08/06/2024 13:57:54.111	Refresh Config	Configuration successfully received! (Expected: 34 Found: 34)
08/06/2024 13:57:54.134	EnlargeBuffer	Buffer size changed to 1024 [True]
08/06/2024 13:57:57.546	EnqueueProgram	Push Header
08/06/2024 13:57:57.547	EnqueueProgram	Push File, 18471 lines
08/06/2024 13:57:57.549	EnqueueProgram	Running program, 18471 lines
08/06/2024 13:57:57.760	SetStatus   	Machine status [Run]
08/06/2024 14:03:11.907	Issue detector	StopResponding [993,-1,-1]
08/06/2024 14:07:02.096	ProgramEnd  	Job Executed: 18471 lines, 0 errors, 9 min,4 sec
08/06/2024 14:07:02.097	EnqueueProgram	Push Footer
08/06/2024 14:07:02.367	SetStatus   	Machine status [Idle]

Aqui o comlog:
00000	00800327	log	Recording session started @ 08/06/2024 13:52:30.759 - Session ID [a7f121e1]
00001	00800793	tx	[3F]
00002	00800798	rx	<Idle|MPos:0.000,0.000,0.000>
00003	00801293	tx	[3F]
00004	00801304	rx	<Idle|MPos:0.000,0.000,0.000>
00005	00801797	tx	[3F]
00006	00801800	rx	<Idle|MPos:0.000,0.000,0.000>
00007	00802293	tx	[3F]
00008	00802297	rx	<Idle|MPos:0.000,0.000,0.000>
00009	00802795	tx	[3F]
00010	00802803	rx	<Idle|MPos:0.000,0.000,0.000>
00011	00803291	tx	[3F]
00012	00803298	rx	<Idle|MPos:0.000,0.000,0.000>
01071	01068292	tx	[3F]
01072	01068293	rx	<Idle|MPos:0.000,0.000,0.000>
01073	01068794	tx	[3F]
01074	01068800	rx	<Idle|MPos:0.000,0.000,0.000>
01075	01069295	tx	[3F]
01076	01069306	rx	<Idle|MPos:0.000,0.000,0.000>
01077	01069793	tx	[3F]
01078	01069801	rx	<Idle|MPos:0.000,0.000,0.000>
01079	01070292	tx	[3F]
01080	01070297	rx	<Idle|MPos:0.000,0.000,0.000>
01081	01070792	tx	[3F]
01082	01070793	rx	<Idle|MPos:0.000,0.000,0.000>
01083	01071296	tx	[3F]
01084	01071299	rx	<Idle|MPos:0.000,0.000,0.000>
01085	01071792	tx	[3F]
01086	01071795	rx	<Idle|MPos:0.000,0.000,0.000>
01087	01072292	tx	[3F]
01088	01072302	rx	<Idle|MPos:0.000,0.000,0.000>
01089	01072794	tx	[3F]
01090	01072797	rx	<Idle|MPos:0.000,0.000,0.000>
01091	01073296	tx	[3F]
01092	01073304	rx	<Idle|MPos:0.000,0.000,0.000>
01093	01073792	tx	[3F]
01094	01073800	rx	<Idle|MPos:0.000,0.000,0.000>
01095	01074291	tx	[3F]
01096	01074295	rx	<Idle|MPos:0.000,0.000,0.000>
01097	01074793	tx	[3F]
01098	01074802	rx	<Idle|MPos:0.000,0.000,0.000>
01099	01075293	tx	[3F]
01100	01075299	rx	<Idle|MPos:0.000,0.000,0.000>
01101	01075791	tx	[3F]
01102	01075794	rx	<Idle|MPos:0.000,0.000,0.000>
01103	01076292	tx	[3F]
01104	01076302	rx	<Idle|MPos:0.000,0.000,0.000>
01105	01076483	com	Close Emulator [USER]
01106	01076489	com	Close Emulator [CORE]
01107	01078579	log	Recording session stopped @ 08/06/2024 13:57:09.010 - Session ID [a7f121e1]

**Comment by Tribo-2dot0** [comment]
Aqui um print do momento da longa pausa (a que o Telegram avisa como Parou de Responder):
![Captura de tela 2024-06-08 140312](https://github.com/arkypita/LaserGRBL/assets/83982525/669f0475-4833-4d45-9409-d8efe66b73b0)

**Comment by arkypita** [comment]
It's normal. The emulator does not handle sending the "interpolated" cursor position during a long movement, it only transmits it at the start and end position of the segment. When the emulator has to execute a movement between two points only send the first point, then delay until the time needed by the movement, then send the final position.

This causes the LaserGRBL control mechanism that monitors the position sent by the machine to change to think that the machine is no longer performing the movement/is no longer responding. It's a problem that exclusively concerns the emulator and that I don't think I want to put time solving.

**Comment by arkypita** [comment]
PS: was the same in all version of LaserGRBL, it is not an issue of v7.x

**Reply by Tribo-2dot0** [reply]
> PS: was the same in all version of LaserGRBL, it is not an issue of v7.x

É que nunca antes tinha pausado por tanto tempo e nem recebido aviso do Telegram... testei com v6.2.2 e aconteceu o mesmo, até durante a gravação raster, mas parece que é só com esse design específico, vou refaze-lo, com outros projetos ocorre tudo certo.
